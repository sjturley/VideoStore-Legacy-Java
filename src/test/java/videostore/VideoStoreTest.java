package videostore;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.QuietReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class VideoStoreTest {
    @Test
    public void approveCustomerStatement() {
        Customer customer = new Customer("Steve") {
            @Override
            protected List<Rental> getRentals() {
                return Arrays.asList(new Rental(new Movie("Star Wars", Movie.REGULAR), 3));
            }
        };
        String statement = customer.statement();

        Approvals.verify(statement);
    }
}
