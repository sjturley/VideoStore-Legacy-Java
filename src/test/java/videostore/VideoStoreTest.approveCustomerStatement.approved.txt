[0, 3] => Rental Record for Steve
	Star Wars	3.5
Amount owed is 3.5
You earned 1 frequent renter points 
